infrastructure/src/main/java/org/corfudb/infrastructure/health/HealthMonitor.java

```diff
@@ -55,6 +83,24 @@ private void addIssue(Issue issue) {
         });
     }
 
+    private void checkLiveness() {
+        ThreadMXBean tmx = ManagementFactory.getThreadMXBean();
+        long[] ids = tmx.findDeadlockedThreads();
```

Comment:
This is more of a question than a comment - Is there a performance impact of invoking this every second?

Reply:
From what I found it does look like an expensive operation, although it's implemented in native code.
https://hg.openjdk.java.net/jdk7/modules/hotspot/file/9646293b9637/src/share/vm/services/management.cpp#l1795
https://github.com/openjdk-mirror/jdk7u-hotspot/blob/master/src/share/vm/runtime/vm_operations.cpp#L245
https://hg.openjdk.java.net/lambda/lambda/hotspot/file/8a63c6323842/src/share/vm/services/threadService.cpp#l290
I will increase the poll to 1 minute for now, which seems reasonable.